backends/advanced/src/advanced_omi_backend/controllers/conversation_controller.py

```
@@ -177,12 +178,87 @@ async def get_conversations(user: User):
 
 
-async def delete_conversation(conversation_id: str, user: User):
-    """Delete a conversation and its associated audio files. Users can only delete their own conversations."""
+async def _soft_delete_conversation(conversation: Conversation, user: User) -> JSONResponse:
+    """Mark conversation and chunks as deleted (soft delete)."""
+    conversation_id = conversation.conversation_id
+
+    # Mark conversation as deleted
+    conversation.deleted = True
+    conversation.deletion_reason = "user_deleted"
+    conversation.deleted_at = datetime.utcnow()
```

Ruff (0.14.10)

Undefined name `datetime`

(F821)